sympy/polys/tests/test_polytools.py

```diff
@@ -1928,6 +1929,18 @@ def test_issue_7864():
     assert r == 0
 
 
+def test_gcdex_steps():
+    f = Poly(x**5 + 2*x**4 - x**2 + 1, x)
+    g = Poly(x**4 - 1, x)
+    eea_result = gcdex_steps(f, g)
+
+    r_degree = 5
+    for si, ti, ri in eea_result:
+        assert si*f + ti*g == ri
+        assert ri.degree() < r_degree
+        r_degree = ri.degree()
```

Comment:
I think this needs a few more tests. At least there should be a test with Expr inputs and a test that uses the domain implementation.